### PR TITLE
issue #27164 - Initial commit of core desktop changes required for Workflow migration project

### PR DESCRIPTION
**Pull request by andersdd**
[DO NOT MERGE]

Should be ready but requires xtuple/xtuple#3020

### CODE REVIEW

File: guiclient/project.ui [hunk 1]
```diff
@@ -785,7 +849,7 @@ to be bound by its terms.</comment>
        <item>
         <widget class="Comments" name="_comments">
          <property name="type">
-          <enum>Comments::Project</enum>
+          <number>0</number>
```
Comment (gilmoskowitz) [inline guiclient/project.ui]: This doesn't look right here and for `_documents`.

File: guiclient/project.ui [hunk 2]
```diff
@@ -901,7 +975,6 @@ to be bound by its terms.</comment>
  </customwidgets>
  <tabstops>
   <tabstop>_number</tabstop>
-  <tabstop>_status</tabstop>
```
Comment (gilmoskowitz) [inline guiclient/project.ui]: The tab order doesn't look right. Status and the new fields are now missing.

File: guiclient/project.cpp [hunk 3]
```diff
@@ -272,26 +276,21 @@ enum SetResponse project::set(const ParameterList &pParams)
     {
       _mode = cView;
       
-      _owner->setEnabled(false);
+      _infoGroup->setEnabled(false);
```
Comment (gilmoskowitz) [inline guiclient/project.cpp]: This assumes that the relevant fields will stay in the QGroupBox, making it fragile.

File: guiclient/task.ui [hunk 4]
```diff
@@ -523,7 +561,7 @@ to be bound by its terms.</comment>
        <item row="0" column="0">
         <widget class="Comments" name="_comments">
          <property name="type">
-          <enum>Comments::Task</enum>
+          <number>0</number>
```
Comment (gilmoskowitz) [inline guiclient/task.ui]: Revert the `type` change here and for `_documents`.

File: guiclient/vendor.cpp [hunk 5]
```diff
@@ -588,6 +594,7 @@ bool vendor::sSave()
           "  <? value(\"vend_pocomments\") ?>,"
           "  <? value(\"vend_fobsource\") ?>,"
           "  <? value(\"vend_fob\") ?>,"
+          "  <? value(\"vend_potype_id\") ?>,"
```
Comment (gilmoskowitz) [inline guiclient/vendor.cpp]: `'` is better than `\"`

File: guiclient/vendor.ui [hunk 6]
```diff
@@ -39,7 +39,7 @@ to be bound by its terms.</comment>
             <bool>false</bool>
            </property>
            <property name="canEdit">
-            <bool>true</bool>
+            <bool>false</bool>
```
Comment (gilmoskowitz) [inline guiclient/vendor.ui]: Is this intentional?

File: guiclient/vendor.ui [hunk 7]
```diff
@@ -1143,7 +1194,7 @@ and Purchase Order amounts</string>
        <item>
         <widget class="Documents" name="_documents">
          <property name="type">
-          <enum>Documents::Vendor</enum>
+          <number>0</number>
```
Comment (gilmoskowitz) [inline guiclient/vendor.ui]: revert

File: guiclient/vendor.ui [hunk 8]
```diff
@@ -1259,7 +1310,7 @@ and Purchase Order amounts</string>
               <enum>Qt::StrongFocus</enum>
              </property>
              <property name="type">
-              <enum>Comments::CRMAccount</enum>
+              <number>0</number>
```
Comment (gilmoskowitz) [inline guiclient/vendor.ui]: revert

File: guiclient/poType.cpp [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+                        " WHERE potype_id <> :potype_id; " );
+    defaultSave.bindValue(":potype_id", _potypeid);
+    defaultSave.exec();
+    if (ErrorReporter::error(QtCriticalMsg, this, tr("Error Updating Purchase Order defaults"),
+                                typeSave, __FILE__, __LINE__))
+    {
+      return;
+    }
+  }
+  omfgThis->sItemGroupsUpdated(-1, true);
```
Comment (andersdd) [inline guiclient/poType.cpp]: The original author just grabbed something to cause an event by the looks

**Pull request merged**